docs/guides/environmental-variables.md

```diff
@@ -0,0 +1,57 @@
+
+For example, you can do the following:
+
+```graphql showLineNumbers
+type Query {
+  users: [User]! @http(baseUrl: "{{env.API_ENDPOINT}}", path: "/users")
+}
+```
+
+Here, the `API_ENDPOINT` is the environmental variable. This value needs to be already set in the device where your server will be running. The value gets picked up by the server when it starts up and is available in the `env` Context variable.
+
+## Security aspects and best practices
+
+Environmental variables help you in mitigating most of the security risks. While this is true, it is also necessary to understand that, it doesn't actually help you in removing the risks completely, since the values are still in plain text. Configurational values might not be as sensitive, but the secrets still do have a chance to be compromised.
+
+To make sure your secrets stay safe and secure all through, here are some tips and tricks that you might consider using for your application:
+
+- **Use a `.env` file** \
+  This is a common practice that is followed by most developers. You can create a `.env` file in the root directory of your project and store all your environmental variables in it. This file should not be committed to your version control system, and should be added to the `.gitignore` file. This way, you can make sure that your secrets are not exposed to the public. Additionally, you can also use a `.env.example` file to list out all the environmental variables that are required for your application to run. This way, you can make sure that all the developers who work on your project are aware of the variables that are required for the application to run.
+
+- **Use secrets in K8s** \
+  If you are using Kubernetes to deploy your application, you can use the [Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) feature to store your environmental variables. This way, you can make sure that your secrets are not exposed to the public, and also, you can make sure that the values are not hardcoded in your codebase. Not to mention, it becomes very easy to change the values when required.
+
+- **Store secrets through the GUI** \
+  If you are using a cloud provider to deploy your application, you can use the GUI provided by the cloud provider to store your environmental variables. They generally provide an intuitive UI, and furthermore, if your application is containerized and scales up and down automatically, this becomes the default go-to option.
+
+- **Using external services** \
+  There are many providers out there, mostly open-sourced softwares, that lift off the burden of security from your shoulders and provide you with a secure way to store your secrets. You can use these services to store your secrets and then access them in your application. Some of the popular ones are [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/), [Hashicorp Vault](https://www.vaultproject.io/), [Keywhiz](https://square.github.io/keywhiz/), etc.
```

Comment:
This whole section is more about managing secrets and less about env variables or tailcall.

Reply:
okay then, should i add more examples in here about where and how we can use environment variables? or do you have something else in mind?